src/arduino/pressure_controller.cpp

```diff
@@ -0,0 +1,254 @@
+    }
+    case CyclePhases::EXHALATION: {
+      exhale();
+      break;
+    }
+    case CyclePhases::HOLD_EXHALATION:
+    {
+        break;
+    }
+    default:
+    {
+        inhale();
+    }
+  }
+
+  safeguards(p_currentCentieme);
+
+  DBG_PHASE_PRESSION(p_currentCentieme, 50, m_phase, m_currentPressure)
+}
+
+void PressureController::updatePhase(uint16_t p_currentCentieme) {
+  if (p_currentCentieme <= m_centiSecPerInhalation) {
+    m_phase = m_currentPressure >= m_peakPressure ? CyclePhases::INHALATION
+                                                  : CyclePhases::PLATEAU;
+  }
+  else
+  {
+    m_phase = CyclePhases::EXHALATION;
+  }
+}
+
+void PressureController::inhale()
+{
+  // Open the air stream towards the patient's lungs
+  m_blower.command = ANGLE_FERMETURE - ANGLE_MULTIPLICATEUR * m_aperture;
+
+  // Open the air stream towards the patient's lungs
+  m_patient.command =
+      ANGLE_FERMETURE + ANGLE_MULTIPLICATEUR * ANGLE_OUVERTURE_MAXI;
+
+  // Update the peak pressure
+  m_peakPressure = m_currentPressure;
+}
+
+void PressureController::plateau()
+{
+  // Deviate the air stream outside
+  m_blower.command =
+      ANGLE_FERMETURE + ANGLE_MULTIPLICATEUR * ANGLE_OUVERTURE_MAXI;
+
+  // Close the air stream towards the patient's lungs
+  m_patient.command = ANGLE_FERMETURE;
+
+  // Update the plateau pressure
+  m_plateauPressure = m_currentPressure;
+}
+
+void PressureController::exhale()
+{
+  // Deviate the air stream outside
+  m_blower.command =
+      ANGLE_FERMETURE + ANGLE_MULTIPLICATEUR * ANGLE_OUVERTURE_MAXI;
+
+  // Open the valve so the patient can exhale outside
+  m_patient.command = m_patient.failsafe;
+
+  // Update the PEEP
+  m_peep = m_currentPressure;
+}
+
+void PressureController::safeguards(uint16_t p_currentCentieme) {
+  // si pression crête > max, alors fermeture blower de 2°
+  if (m_currentPressure > m_maxPeakPressure) {
+    DBG_PRESSION_CRETE(p_currentCentieme, 80)
+    m_blower.command = m_blower.position - 2;
+  }
+  // si pression plateau > consigne param, alors ouverture expiration de 1°
+  if (m_phase == CyclePhases::PLATEAU &&
+      m_currentPressure > m_maxPlateauPressure) {
+    DBG_PRESSION_PLATEAU(p_currentCentieme, 80)
+    m_patient.command = m_blower.position + 1;
+  }
+  // si pression PEP < PEP mini, alors fermeture complète valve expiration
+  if (m_currentPressure < m_minPeep) {
+    DBG_PRESSION_PEP(p_currentCentieme, 80)
+    m_patient.command = ANGLE_FERMETURE;
+    m_phase = CyclePhases::HOLD_EXHALATION;
+  }
+}
+
+void PressureController::computeCentiSecParameters() {
+  m_centiSecPerCycle = 60 * 100 / m_cyclesPerMinute;
+  // Inhalation = 1/3 of the cycle duration, Exhalation = 2/3 of the cycle
+  // duration
+  m_centiSecPerInhalation = m_centiSecPerCycle / 3;
+}
+
+void PressureController::executeCommands()
+{
+    m_blower.execute();
+    m_patient.execute();
+}
+
+void PressureController::onCycleMinus() {
+#ifdef DEBUG
+  Serial.println("nb cycle --");
```

Comment:
This should use `DBG_DO` from `debug.h`.

_This issue happens several times in this file_

Reply:
Fixed with 068cb06f625e922e7cbe366e3a4aba98732ef3d5

Reply:
Done